### PR TITLE
Rename labs and samples project groups

### LINKED ISSUE #1035: projects/labs と projects/samples を _labs / _samples にリネーム
## Overview / 概要

`projects/labs` → `projects/_labs`、`projects/samples` → `projects/_samples` にリネームし、非アクティブなプロジェクト（実験・サンプル）を Active プロジェクト（`projects/<name>`）と視覚的に差別化する。

## Objective / 目的

`projects/` 直下に `_labs` と `_samples` が並ぶことで、Active プロジェクトと非アクティブプロジェクトの区別がディレクトリ一覧上で明確になる。

## Background / 背景

現在の `projects/` 構成:
```
projects/
├── labs/          # 実験・PoC（19プロジェクト）
├── samples/       # 教材・テンプレート・CI検証（29プロジェクト）
├── chatbot-ui/    # Active
├── edit-vid/      # Active
├── file-server/   # Active
├── portfolio/     # Active
├── portal/        # Active
└── simple-agent-poc/  # Active
```

`_` プレフィックスを付けることでソート順でも末尾に集まり、アクティブプロジェクトが目立つようになる。

## Related Issues/PRs / 関連Issue・PR

None

## Implementation Approach / 実装方針

1. ディレクトリをリネーム
2. CI/CD コアロジックの glob パターンを更新（修正必須。修正漏れがあると CI/CD で labs/samples 配下の変更が検出されなくなる）
3. テストスクリプトのパスを更新
4. ドキュメントのパス表記を一括置換

```mermaid
flowchart TD
  A["projects/labs/ → projects/_labs/"] --> C["CICD 変更検出復旧"]
  B["projects/samples/ → projects/_samples/"] --> C
  C --> D["テストスクリプト修正"]
  D --> E["テスト実行・確認"]
  E --> F["ドキュメント修正"]
```

### CI/CD 影響の核心

`get-changed-dirs.sh` の `is_project_root_dir()` は、`projects/labs/<name>` と `projects/samples/<name>` をサブ階層プロジェクトルートとして認識する特殊ロジックを持つ。この glob パターンを更新しなければ、labs/samples 配下の変更が CI/CD で一切検出されなくなる。

```bash
# get-changed-dirs.sh 修正前
is_project_root_dir() {
  local dir="$1"
  [[ "$dir" == projects/* && "$dir" != */*/* ]] && return 0
  [[ "$dir" == projects/labs/* && "$dir" != projects/labs/*/* ]] && return 0
  [[ "$dir" == projects/poc/* && "$dir" != projects/poc/*/* ]] && return 0
  [[ "$dir" == projects/samples/* && "$dir" != projects/samples/*/* ]] && return 0
  return 1
}

# 修正後
is_project_root_dir() {
  local dir="$1"
  [[ "$dir" == projects/* && "$dir" != */*/* ]] && return 0
  [[ "$dir" == projects/_labs/* && "$dir" != projects/_labs/*/* ]] && return 0
  [[ "$dir" == projects/poc/* && "$dir" != projects/poc/*/* ]] && return 0
  [[ "$dir" == projects/_samples/* && "$dir" != projects/_samples/*/* ]] && return 0
  return 1
}
```

### 修正対象一覧

| 優先度 | ファイル | 修正内容 |
|--------|----------|----------|
| 高 | `.github/actions/get-changed-directories/get-changed-dirs.sh` | L39, L41 の glob パターン |
| 高 | `.github/actions/get-changed-directories/test-project-root.sh` | L27, L29 + 全テストケースのパス (L99-225) |
| 高 | `.github/actions/get-changed-projects/test-poc-exclude.sh` | L131-134, L174-180 のテストパス |
| 中 | `docs/projects-classification.md` | `projects/labs` → `projects/_labs` など全パス表記 |
| 中 | `projects/README.md` | `projects/labs/<name>` → `projects/_labs/<name>` など |
| 中 | `.github/actions/get-changed-directories/README.md` | サンプルパス |
| 中 | `.github/actions/get-changed-projects/README.md` | サンプルパス |
| 低 | `.claude/skills/dockerfile-designer/SKILL.md` | 参照パス |
| 低 | `projects/labs/*/README.md` (3ファイル) | プロジェクト内 cd/path コマンド |
| 低 | `projects/labs/execution-sandbox/docs/*.md` (2ファイル) | ドキュメント内パス |
| 低 | `projects/samples/node-mcp-server/README.md` | プロジェクト内パス |

### 影響なし

- Workflow YAML（`pullrequest-check.yml`, `docker-build.yml`）— actions に委譲しているため直接修正不要
- Python / TypeScript ソースコード — import パスとしての参照は 0 件
- `package.json`, `Dockerfile`, `docker-compose.yaml` — パス参照なし

## Tasks / タスク

- [ ] ディレクトリリネーム: `mv projects/labs projects/_labs`
- [ ] ディレクトリリネーム: `mv projects/samples projects/_samples`
- [ ] `get-changed-dirs.sh` の `is_project_root_dir()` glob パターン修正
- [ ] `test-project-root.sh` の関数定義 + 全テストケースパス修正
- [ ] `test-poc-exclude.sh` のテストパス修正
- [ ] `docs/projects-classification.md` パス表記修正
- [ ] `projects/README.md` パス表記修正
- [ ] `.github/actions/get-changed-directories/README.md` サンプルパス修正
- [ ] `.github/actions/get-changed-projects/README.md` サンプルパス修正
- [ ] `.claude/skills/dockerfile-designer/SKILL.md` 参照パス修正
- [ ] 各プロジェクト内 README パス修正 (6ファイル)
- [ ] テストスクリプト実行し全件パスを確認

## Out of Scope / スコープ外

- Workflow YAML ファイルの修正（actions に委譲しているため不要）
- 各プロジェクトの実装内容変更
- 新規プロジェクトの分類変更（既存分類を維持したままのリネームのみ）

### Validation Policy / 検証方針

`projects/` 配下の CI/CD 変更検出はシェルスクリプトのユニットテストで保護されている。修正後に両方のテストを実行し全件パスすることを確認する。

### Documentation / ドキュメント

`docs/projects-classification.md` と `projects/README.md` のパス表記を更新する。

## Acceptance Criteria / 完了条件

- [ ] `projects/_labs/` と `projects/_samples/` が存在し、旧ディレクトリが存在しない
- [ ] `test-project-root.sh` が全テストケース成功
- [ ] `test-poc-exclude.sh` が全テストケース成功
- [ ] 全ドキュメントで新しいパス表記になっている

## Test Perspectives / テスト観点

| Area / 観点 | What to Verify / 確認内容 | Method / 確認方法 |
| --- | --- | --- |
| Normal path / 正常系 | `projects/_labs/<name>` 内の変更が changed_dirs.txt に検出される | `test-project-root.sh` 実行 |
| Normal path / 正常系 | `projects/_samples/<name>` 内の変更が changed_dirs.txt に検出される | `test-project-root.sh` 実行 |
| Edge cases / 境界条件 | PoC ではない `_labs` / `_samples` プロジェクトがビルド対象に含まれる | `test-poc-exclude.sh` 実行 |
| Regression / 回帰 | 既存の Active プロジェクト検出が影響を受けない | `test-project-root.sh` 実行 |
| Operations / 運用 | CI/CD (pullrequest-check, docker-build) が正常動作する | PR 作成後の CI 実行確認 |

## Notes / 補足

`projects/poc/` は現在空ディレクトリだが、`get-changed-dirs.sh` に glob パターンが残っている。この Issue のスコープ外だが、別途整理候補。

### PR DESCRIPTION
**Pull request by u7chan**
## Issues

- Close #1035

## Why

`projects/labs` and `projects/samples` contain inactive experiment and sample projects. Renaming them to underscore-prefixed directories makes these groups visually distinct from active projects under `projects/<name>`.

## Summary

This PR renames `projects/labs` to `projects/_labs` and `projects/samples` to `projects/_samples`. It also updates CI/CD change detection patterns, related shell tests, and documentation references to use the new paths.

## Changes

- Rename labs and samples project group directories with `git mv`.
- Update `get-changed-dirs.sh` project-root detection for `_labs` and `_samples`.
- Update shell tests and README examples for the new paths.
- Update project classification documentation and project-local README path references.

## Verification

- `bash .github/actions/get-changed-directories/test-project-root.sh` - passed
- `bash .github/actions/get-changed-projects/test-poc-exclude.sh` - passed
- `git diff --cached --check` - passed before commit
- `rg -n "projects/(labs|samples)" .github docs projects .claude -g '!**/node_modules/**' -g '!**/dist/**' -g '!**/static/tailwindcss*.js'` - no matches
- `git diff --diff-filter=R --name-status HEAD origin/main | awk '$1 ~ /^R/ {print $2}' | rg '^projects/_(labs|samples)/' | head -n 20` - confirmed renamed paths are emitted

### CODE REVIEW

File: docs/projects-classification.md [hunk 1]
```diff
@@ -9,9 +9,9 @@
 
 | Category | Path | Rule | Examples |
 | --- | --- | --- | --- |
-| main | `projects/<project>` | deploy 対象、または継続保守するアプリ/ライブラリ | `chatbot-ui`, `edit-vid`, `file-server`, `portfolio`, `portal`, `simple-agent-poc` |
-| labs | `projects/labs/<project>` | 今後も育てる、または再利用する可能性がある実験。PoC もここに含める | `backend-sse-chat-poc`, `hono-react-inertia`, `long-running-mcp-server`, `threejs-obj-viewer` |
-| samples | `projects/samples/<project>` | 教材、テンプレート、CI/CD 検証、公式サンプル改造 | `fastapi-api-base`, `gemini-sdk-example`, `react-markdown-preview` |
+| main | `projects/<project>` | deploy 対象、または継続保守するアプリ/ライブラリ | `edit-vid`, `file-server`, `portfolio`, `portal`, `raw-webgl2-shader`, `simple-agent-poc` |
```
Comment (u7chan) [inline docs/projects-classification.md]: [ask] ❓ この行ではパスのリネームに加えて、`chatbot-ui` を main から labs へ移動し、`raw-webgl2-shader` を main に追加しています。これらは分類台帳の変更であり、PR説明にある「labs/samples のリネーム」のスコープ外に見えます。

同様の変更が L13（Examples 列）、L44（main 台帳）、L45（labs 台帳）にもあります。これらは意図的なものですか？そうであれば PR 説明に追記したほうがよさそうです。意図的でなければ、別PRに切り出したほうが差分が明確になります。

**Review by u7chan — COMMENTED** [review]
AIレビュー補助（Codex / deepseek-v4-pro）によるコメントです 🤖

File: docs/projects-classification.md (hunk 1 above)
Comment (u7chan) [reply docs/projects-classification.md]: 対応しました。

分類台帳はメンテナンスで実体とズレやすいため、`## 現在の分類台帳` を削除しました。あわせて `Examples` 列も個別プロジェクト名の固定リストになるため削除し、分類ルールは `Category / Path / Rule` に絞っています。

Commit: ce6ad6c5

検証:
- `git diff --check` passed
- `rg -n "projects/(labs|samples)|現在の分類台帳|Examples" docs/projects-classification.md` no matches
- `bash .github/actions/get-changed-directories/test-project-root.sh` passed
- `bash .github/actions/get-changed-projects/test-poc-exclude.sh` passed

Comment (u7chan) [reply docs/projects-classification.md]: 改善を確認しました ✅

- `Examples` 列削除 → 個別プロジェクト名の固定リスト混入が解消
- `## 現在の分類台帳` 削除 → 台帳メンテのズレリスクも解消
- Category / Path / Rule のみの分類ルールとして簡潔に

AIレビュー補助（Codex / deepseek-v4-pro）による再チェックコメントです 🤖
